internal/controllers/machinedeployment/machinedeployment_sync.go

```diff
@@ -540,7 +539,7 @@ func calculateStatus(allMSs []*clusterv1.MachineSet, newMS *clusterv1.MachineSet
 			V1Beta1: &clusterv1.MachineDeploymentV1Beta1DeprecatedStatus{
 				Conditions:          conditions,
 				UpdatedReplicas:     mdutil.GetActualReplicaCountForMachineSets([]*clusterv1.MachineSet{newMS}),
```

Comment:
l.554 Q: Should we really still use v1beta1 counters to calculate .status.phase?

Reply:
Bonus if we can make the phase and conditions line up (e.g. for scaling, not sure if more)

Reply:
No we should not.

Is that ok if I will address this in a follow up PR, I have point in my todo list to look into old UpdateStatus funcs (e.g. calculateV1Beta1Status in MD,MS - reconcileV1Beta1Status ) and move out everything not deprecated, like .status.phase.

Reply:
Absolutely fine to address in a follow-up, just noticed it here during review